### DISCUSSION TITLE
Need explanation of the colors on the timeline when zoomed out

### DISCUSSION THREAD

**Discussion by Tagirijus** [almarklein/timetagger#466] [Q&A] [answered]
According to another maybe related issue (#465) - I wondered, how exactly the order logic of the colors in the timeline is. When you zoom out at some point you get a more graphical representation of the amount of tags (and their assigned color) used, if I understand it correct. What still is not clear to me in this representation is:

- What is the logic of the order? Is it similar to the reply Almar gave me int he linked issue and it's about the _amount_ of entries?
- What is it about the dimmed area and the more bright one? I guess it has something to do with the duration for e.g. that day or so. Higher durations for a day have less dimmed area, for example. Yet it's still not 100% clear to me.

If there is too much time: Maybe the docs could be extended to explain this as well? (;

**Comment by almarklein** [comment, marked as answer]
> What is the logic of the order? 

In the week/month/year statistics shown in the timeline, each color represents a tag, and they are simply sorted by total duration.

https://github.com/almarklein/timetagger/blob/3a381b0097c8d65feb87b7956f4d32f56aba1d1b/timetagger/app/front.py#L2534-L2536

> What is it about the dimmed area and the more bright one?

It is an indication of how much time was spent compared to the total time in that period. I had to look in the source code to see how its actually implemented. It is not a linear relation, because then it'd be just a thin bar most of the time.

**Reply by Tagirijus** [reply]
Ah thanks for the explanation and for digging into the code! (=

**Reply by almarklein** [reply]
Looking a bit closer: the non-linear part is because of you have records that contain two tags, the time for that record contributes twice to growing the more saturated area.
